### PR TITLE
157 dmarc config identified

### PR DESCRIPTION
**Pull request by markXLIX**

### CODE REVIEW

File: src/validations/rules/ssp.sch [hunk 1]
```diff
@@ -2246,14 +2246,33 @@
...
                 diagnostics="has-this-system-component-diagnostic"
                 doc:guide-reference="Guide to OSCAL-based FedRAMP System Security Plans §5.4.6"
                 id="has-this-system-component"
                 role="error"
                 test="exists(oscal:component[@type eq 'this-system'])">A FedRAMP SSP must have a self-referential (i.e., to the SSP itself)
                 component.</sch:assert>
+            <sch:let
+                name="emailString"
+                value="'email', 'e-mail', 'electronic mail'" />
+            <sch:report
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:2260]: Why is this a `report` rather than an `assert`?

File: src/validations/rules/ssp.sch [hunk 2]
```diff
@@ -2246,14 +2246,33 @@
...
             <sch:assert
                 diagnostics="has-this-system-component-diagnostic"
                 doc:guide-reference="Guide to OSCAL-based FedRAMP System Security Plans §5.4.6"
                 id="has-this-system-component"
                 role="error"
                 test="exists(oscal:component[@type eq 'this-system'])">A FedRAMP SSP must have a self-referential (i.e., to the SSP itself)
                 component.</sch:assert>
+            <sch:let
+                name="emailString"
+                value="'email', 'e-mail', 'electronic mail'" />
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:2259]: I am uncertain this will result in a sequence.

File: src/validations/rules/ssp.sch [hunk 3]
```diff
@@ -2246,14 +2246,33 @@
...
+            <sch:report
+                diagnostics="has-email-diagnostic"
+                id="has-email"
+                role="warning"
+                test="
+                    if (//*[lower-case(.) = $emailString])
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:2265]: The document
```
<ssp>
    <p>email is used.</p>
</ssp>
```
assessed using the XPath expression
```
//*[lower-case(.) = ('email', 'e-mail', 'electronic mail')]
```
will be considered to not have the word "email".

The document
```
<ssp>
    <p>email</p>
</ssp>
```
would. But both documents mention email.

**Comment by GaryGapinski** [comment]
The current assertion message is "_This FedRAMP SSP has references to 'email', 'e-mail', or 'electronic mail'._". That message is incorrect if there is no mention of electronic mail in the SSP.

I think the assertion message should somehow express the two factors contributing to a positive outcome, namely

1. The SSP does not mention electronic mail, or
2. The SSP mentions electronic mail and also mentions SPF, DKIM, and DMARC

as opposed to asserting just the second. Perhaps something like "_Electronic mail is either absent or sufficiently specified._".

The current diagnostic message is _"DMARC, SPF, and DKIM is referenced in this SSP._". That message is incorrect when the assertion fails.

If the assertion fails, an "analogous negation" message would be something like "_Electronic mail is present but appears to lack documentation of SPF, DKIM, and DMARC._".

**Comment by markXLIX** [comment]
My reading of the task was that email without DMARC, etc was the only desired hit.  I worded the message as the positive because I thought that was the desired output.  The user only wanted to know when these all occurred as stated so they did not need to go looking for them.

I am in the process of adjusting the code to match your reading.

File: src/validations/rules/ssp.sch [hunk 4]
```diff
@@ -2396,9 +2396,9 @@
                         ../oscal:control-implementation/oscal:implemented-requirement[@control-id eq 'si-8']//*[matches(., 'SPF')] and
                         ../oscal:control-implementation/oscal:implemented-requirement[@control-id eq 'si-8']//*[matches(., 'DKIM')])
                         then
-                            (true())
+                            (false())
                         else
-                            (false()))
+                            (true()))
                     else
                         (true())">This FedRAMP SSP references 'email', 'e-mail', or 'electronic mail'.</sch:assert>
```
Comment (GaryGapinski) [inline src/validations/rules/ssp.sch:2403]: This positive test result assertion message (**_This FedRAMP SSP references 'email', 'e-mail', or 'electronic mail'._**)remains incorrect. It does not address the absence within the SSP of any mention of email.

Please see previous comments regarding positive test outcome assertion messages.

**Review by GaryGapinski — CHANGES_REQUESTED** [review]

**Review by GaryGapinski — APPROVED** [review]
Looks good to go.

**Pull request merged**